### PR TITLE
Volume Group Resource Update

### PR DESCRIPTION
**Pull request by prajwalpatil25**
# Description
Implemented volume group resource update operation.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [X] Acceptance tests
image

### CODE REVIEW

File: powerstore/resource_volume_group.go [hunk 1]
```diff
@@ -236,6 +236,144 @@ func (r *resourceVolumeGroup) Read(ctx context.Context, req resource.ReadRequest
...
+		for i := 0; i < len(stateVolumeIds); i++ {
+			stateVolumeIdsMap[stateVolumeIds[i]] = stateVolumeIds[i]
+		}
+	}
+
+	removeVolumeIdsMap := make(map[string]string)
+	for i := 0; i < len(stateVolumeIds); i++ {
+		_, found := planVolumeIdsMap[stateVolumeIds[i]]
+		if !found {
+			log.Printf("Elem not found in state")
```
Comment (shenda1) [inline powerstore/resource_volume_group.go]: Elem -> volume

File: powerstore/resource_volume_group.go [hunk 2]
```diff
@@ -236,6 +236,144 @@ func (r *resourceVolumeGroup) Read(ctx context.Context, req resource.ReadRequest
...
+	removeVolumeIdsSlice := []string{}
+	for _, volumeID := range removeVolumeIdsMap {
+		removeVolumeIdsSlice = append(removeVolumeIdsSlice, volumeID)
+	}
+
+	addVolumeIdsMap := make(map[string]string)
+	for i := 0; i < len(planVolumeIds); i++ {
+		_, found := stateVolumeIdsMap[planVolumeIds[i]]
+		if !found {
+			log.Printf("Elem not found in plan")
```
Comment (shenda1) [inline powerstore/resource_volume_group.go]: Elem -> volume

File: powerstore/resource_volume_group.go [hunk 3]
```diff
@@ -236,6 +236,144 @@ func (r *resourceVolumeGroup) Read(ctx context.Context, req resource.ReadRequest
...
+	removeVolumeIdsMap := make(map[string]string)
+	for i := 0; i < len(stateVolumeIds); i++ {
+		_, found := planVolumeIdsMap[stateVolumeIds[i]]
+		if !found {
+			log.Printf("Elem not found in state")
+			removeVolumeIdsMap[stateVolumeIds[i]] = stateVolumeIds[i]
+		}
+	}
+
+	removeVolumeIdsSlice := []string{}
```
Comment (shenda1) [inline powerstore/resource_volume_group.go:369]: Please add the comments for readability purposes. Applicable for all such instances.

File: powerstore/resource_volume_group_test.go [hunk 4]
```diff
@@ -84,6 +84,309 @@ func TestAccVolumeGroup_CreateWithInvalidVolume(t *testing.T) {
...
+					resource.TestCheckResourceAttr("powerstore_volumegroup.test", "name", "test_volume_group"),
+					resource.TestCheckResourceAttr("powerstore_volumegroup.test", "description", "Creating Volume Group"),
+					resource.TestCheckResourceAttr("powerstore_volumegroup.test", "is_write_order_consistent", "false"),
+				),
+			},
+			{
+				Config: ProviderConfigForTesting + VolumeGroupParamsUpdateAddVolume,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("powerstore_volumegroup.test", "name", "test_volume_group"),
+					resource.TestCheckResourceAttr("powerstore_volumegroup.test", "description", "Updating Volume Group"),
```
Comment (shenda1) [inline powerstore/resource_volume_group_test.go:296]: Check for all the parameters. Volume is missing here. Applicable for all such instances.

File: powerstore/resource_volume_group.go (hunk 1 above)
Comment (prajwalpatil25) [reply powerstore/resource_volume_group.go]: done

File: powerstore/resource_volume_group.go (hunk 2 above)
Comment (prajwalpatil25) [reply powerstore/resource_volume_group.go]: done

File: powerstore/resource_volume_group.go (hunk 3 above)
Comment (prajwalpatil25) [reply powerstore/resource_volume_group.go:369]: done

File: powerstore/resource_volume_group_test.go (hunk 4 above)
Comment (prajwalpatil25) [reply powerstore/resource_volume_group_test.go:296]: done

**Comment by AnikaAgiwal2711** [comment]
LGTM

**Comment by AnikaAgiwal2711** [comment]
Update the Branch and then I'll approve

**Review by AnikaAgiwal2711 — DISMISSED** [review]

**Review by shenda1 — DISMISSED** [review]

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Review by Krunal-Thakkar — APPROVED** [review]

**Review by shenda1 — APPROVED** [review]

**Pull request merged**